src/Mongator/Extension/Core.php

```diff
@@ -1148,7 +1148,7 @@ private function parseAndCheckAssociationClass(&$association, $name)
 
     private function mapArrayKeyWithDefault($array, $key, array $mapCallback, $default)
     {
-        if (property_exists($array, $key)) {
+        if (array_key_exists($key, $array)) {
```

Comment:
What was deprecated was usage on objects, not arrays, thus this may help:
`if ((is_array($array) && array_key_exists($key, $array)) || (is_object($array) && property_exists($array::class, $key))) {`

Reply:
@dberstein what about simply `isSet($array[$key])` - didn't you say that fixes it? and it is a bit cleaner

Reply:
@ymalen  `isset` is not clear enough, if mixes key exists and non-nullness of value. On the other hand array_key_exists does exactly that: it tells if a key exists, regardless of its value.

Reply:
Simplified example where key exists (key "null") and  `isset` != `array_key_exists`:

````
$ php --version; php -r '$a=["null"=>null]; var_dump(isset($a["null"]), array_key_exists("null", $a));'
PHP 7.4.10 (cli) (built: Sep  3 2020 18:19:30) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Xdebug v2.9.7, Copyright (c) 2002-2020, by Derick Rethans
    with Zend OPcache v7.4.10, Copyright (c), by Zend Technologies
Command line code:1:
bool(false)
Command line code:1:
bool(true)
````